tests/test_metadataset_v2.py

```diff
@@ -473,6 +477,144 @@ def test_joined_metadataset_joiner(self):
         # Every item must occurr 2 times (2*55).
         assert Counter(txt1_order).most_common(1)[0][1] == 2
 
+    def test_metadataset_fixed_epochs(self):
+        torch.manual_seed(42)
+        worker_config = WorkerConfig(
+            rank=0,
+            world_size=1,
+            num_workers=0,
+            seed_offset=42,
+        )
+
+        # Create a joined dataset configuration
+        fixed_epochs_mds_path = self.dataset_path / "metadataset_fixed_epochs.yaml"
+        with open(fixed_epochs_mds_path, "w") as f:
```

Comment:
Test 2x ds1, 3x ds2 and verify that stop iteration is working correctly, with save/restore